crates/kitsune_p2p/fetch/src/respond.rs

```diff
@@ -78,7 +82,10 @@ impl<C: FetchResponseConfig> FetchResponseQueue<C> {
             let _byte_permit = byte_permit;
 
             let _c_permit = match c_limit.acquire_owned().await {
-                Err(_) => return,
+                Err(_) => {
+                    tracing::error!("Unexpected closed semaphore for concurrent_send_limit");
```

Comment:
This would have failed silently if the semaphore was closed. I prefer to at least know that has happened if it ever does.